src/librustdoc/clean/cfg.rs

```diff
@@ -284,13 +284,24 @@ fn write_with_opt_paren<T: fmt::Display>(
     Ok(())
 }
 
+fn dedup_arr<T: PartialEq>(s: &[T]) -> Vec<&T> {
+    let mut s = s.iter().collect::<Vec<_>>();
+    let mut i = 0;
+    while i < s.len() {
+        let tmp = s[i];
+        s.dedup_by_key(|a| a == &tmp);
+        i += 1;
+    }
+    s
+}
```

Comment:
I don't understand this algorithm and suspect it's not correct (`dedup_by_key` is `dedup` but it compares something other than the value itself, in your case it dedups consecutive values which are both `true` or both `false` in the result of your closure).

I suspect you meant `retain` instead of `dedup_by_key`. However, it's still quadratic.

If you don't care about the order, you can use `v.sort(); v.dedup();`.
If you care about the order, you can use something more like this:
```rust
let mut set = FxHashSet::new();
v.retain(|x| set.insert(x));
```
Or since you start from a slice:
```rust
let mut set = FxHashSet::new();
s.iter().filter(|x| set.insert(x)).collect()
```

None of the above suggestions is quadratic, and they're all simpler.

Reply:
I didn't want to change the order but if you're fine with it then so be it! (it'll make the code simpler so yeay! \o/)

Reply:
> I didn't want to change the order

I provided one example that changes the order and two that don't. The last one is probably what you should use.

Note that I'm only using the `FxHashSet` to check if the value has been seen already, but the set isn't iterated afterward, so the original order is kept.